src/components/MermaidDash/components/DownloadGFCRModal.jsx

```diff
@@ -0,0 +1,224 @@
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown)
+    }
+  }, [handleClose])
+
+  const title = useMemo(() => {
+    const titles = {
+      'no data': downloadModal.noGFCRDataTitle,
+      success: downloadModal.successTitle,
+      failure: downloadModal.failureTitle,
+    }
+
+    return titles[modalMode] || downloadModal.downloadGFCRTitle
+  }, [modalMode])
+
+  const handleSendEmailWithLinkSubmit = async () => {
+    try {
+      const token = isAuthenticated ? await getAccessTokenSilently() : ''
+
+      const projectsToEmail = projectsWithGFCRData.map(({ project_id }) => project_id)
+
+      const reportEndpoint = `${import.meta.env.VITE_REACT_APP_AUTH0_AUDIENCE}/v1/reports/`
+      const requestData = {
+        report_type: 'gfcr',
+        project_ids: projectsToEmail,
+      }
+
+      const response = await fetch(reportEndpoint, {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestData),
+      })
+
+      if (!response.ok) {
+        setErrorMessage(downloadModal.failureContent)
+        setModalMode('failure')
+        return
+      }
+
+      setModalMode('success')
+    } catch {
+      setErrorMessage(downloadModal.failureContent)
+      setModalMode('failure')
+    }
+  }
+
+  const projectListWithGFCRData = projectsWithGFCRData.map((project) => {
+    return (
+      <li key={project.project_id}>
+        {project.project_name}{' '}
+        {userIsMemberOfProject(project.project_id, mermaidUserData) && (
+          <MuiTooltip
+            title={tooltipText.yourProject}
+            placement="top"
+            bgColor={theme.color.primaryColor}
+            tooltipTextColor={theme.color.white}
+          >
+            <IconButton>
+              <IconUserCircle />
+            </IconButton>
+          </MuiTooltip>
+        )}
+      </li>
+    )
+  })
+
+  const downloadContent = (
+    <>
+      <StyledOverflowList>{projectListWithGFCRData}</StyledOverflowList>
+      {projectsWithoutGFCRDataCount > 0 && (
+        <StyledWarningText>
+          <IconInfo />{' '}
+          <span>
+            {pluralize(
+              projectsWithoutGFCRDataCount,
+              'other filtered project does',
+              'other filtered projects do',
+            )}{' '}
+            not have GFCR indicators
+          </span>
+        </StyledWarningText>
+      )}
+    </>
+  )
+
+  const successContent = (
+    <>
+      <p>An email has been sent to {mermaidUserData.email}</p>
+      <p>This can sometimes take up to 15 minutes</p>
+    </>
+  )
+
+  const content = (
+    <ModalBody>
+      {modalMode === 'no data' && <p>{downloadModal.noGFCRDataContent}</p>}
+      {modalMode === 'download' && downloadContent}
+      {modalMode === 'success' && successContent}
+      {modalMode === 'failure' && <p>{errorMessage}</p>}
+    </ModalBody>
+  )
```

Comment:
To avoid excessive conditionals and improve readability, sometimes I use an object to hold the value I want and invoke it like this:
```
const MODAL_CONTENTBY_MODE: {
mode1: <>foo</>,
mode2: <>bar</>,
}
...
<ModalBody> {MODAL_CONTENT_BY_MODE[modalMode]}</ModalBody>

Reply:
good idea! I change both regular and GFCR modals